src/components/CodeSnippet/CodeSnippet.js

```diff
@@ -1,43 +1,159 @@
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { Component } from 'react';
 import classNames from 'classnames';
+import Copy from '../Copy';
 import CopyButton from '../CopyButton';
+import Icon from '../Icon';
 
-const CodeSnippet = ({
-  className,
-  type,
-  children,
-  feedback,
-  onClick,
-  wrappedContentRef,
-  ...other
-}) => {
-  const snippetType =
-    type === 'terminal' ? 'bx--snippet--terminal' : 'bx--snippet--code';
-  const wrapperClasses = classNames('bx--snippet', className, snippetType);
-  return (
-    <div className={wrapperClasses} {...other}>
+export default class CodeSnippet extends Component {
+  static propTypes = {
+    /**
+     * The type of code snippet
+     * can be inline, single or multi
+     */
+    type: PropTypes.string,
+    className: PropTypes.string,
+    children: PropTypes.string,
+    feedback: PropTypes.string,
+    copyLabel: PropTypes.string,
+    onClick: PropTypes.func,
+    /**
+     * Used with multi snippet only
+     * when text is more than 15 lines
+     */
+    showMoreText: PropTypes.string,
+    showLessText: PropTypes.string,
+    /**
+     * Used with inline snippet only
+     * to display alternate color
+     */
+    light: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    type: 'single',
+    showMoreText: 'Show more',
+    showLessText: 'Show less',
+  };
+
+  state = {
+    showBtn: false,
+    expandedCode: false,
+  };
+
+  componentDidMount = () => {
```

Comment:
Can just have `cDM` as a regular class method since it already has access to instance 👍 

```diff
-  componentDidMount = () => {
+  componentDidMount() {
```